bbp/modules/default.nix

```diff
@@ -741,6 +741,26 @@ let
             conflicts = conflicts-modules;
         };
 
+        connectome-tools = pkgs.envModuleGen rec {
+            name = "connectome-tools";
+            moduleFilePrefix = "nix/nse";
+            description = "connectome-tools module generated by nix";
+            packages = [
+              pkgs.connectome-tools
+            ];
+            conflicts = conflicts-modules;
+        };
+
+        psp-validation = pkgs.envModuleGen rec {
+            name = "psp-validation";
+            moduleFilePrefix = "nix/nse";
+            description = "psp-validation module generated by nix";
+            packages = [
+              pkgs.psp-validation
```

Comment:
At first glance the module seems to work fine even without 'getPyModRec' call.
@adevress, @tristan0x, could you please clarify why it's needed?

Reply:
getPyModRec will transform a python package to a list of the python package + all its dependencies recursively.
Other said, it will make the environment module expose not only your python package, but also all the dependencies it need.

Reply:
And if I'm fine with not exposing the dependencies but would prefer them to be loaded explicitly if needed (i.e. don't want to offer transitive dependencies), is it OK?
Or does it go against some convention / breaks something?

Reply:
I would advise you then to:
- Use getPyModRec
- If you want a module to not expose some transitive dependencies, you can remove them from "propagatedBuildInputs" and place them in "buildInputs". They will not be exposed.
- However, by convention, we generally expose every non-bbp python dependencies in a transitive way. Or it becomes quickly an import nightmare for the user.

Reply:
Sorry, still don't get it.
If I do not `getPyModRec`, what would be the effect on, for example, `nix/nse/psp-validation` module?
Do I get it right, that all the scripts that it exposes, would work fine (i.e. they will find all the dependencies they need, their dependencies etc); but "import bluepy" would not be available?

Reply:
> but "import bluepy" would not be available for the client code?

- If bluepy is part of the "propagatedBuildInputs" it will
- if bluepy is part of "buildInputs", it will not.

Reply:
That I understand.
What I still do not grasp, is why 'getPyModRec' is needed.
Anyway, we can clear it up later.

Reply:
If your Python package requires dependencies to run, why don't you want to provide them to the user? Offering the user the choice to load whatever he wants is like giving a child a loaded shotgun.

If you do not use `getPyModeRec` to create a module containing both the Python package and its dependencies, the module will *only* provides the Python package and will most likely be unusable as it is. Besides the user will have to figure by itself the dependency tree to install on the side to use your module. And I don't even mention the issues when mixing Python native package not compiled by Nix...

Reply:
Repeating my question above:
> Do I get it right, that all the scripts that it exposes, would work fine (i.e. they will find all the dependencies they need, their dependencies etc); but "import bluepy" would not be available for the client code?

A child playing with a loaded shotgun have found that _scripts_  exposed by the package (i.e. commands available in `bin`) work fine without getPyModeRec and was confused by that fact.
Now as I checked it myself, I see that it is due to fancy `addsitedir` mangling that happens in the beginning of the wrapper provided for each script.

The lesson I took out is "definitely use `getPyModeRec` if module is intended to be used as a library; up to you if it is mainly an application" OR "always use `getPyModeRec` to avoid bothering about the difference between library and application".

That's the explanation I was looking for, not the funny metaphors.
Thanks for trying though, I guess I have not put my question clearly enough.